src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -172,29 +172,148 @@ public class ChangeManagement implements Serializable {
+
+        if (!deployConfigFile) {
+            useConfigFile = false
+            noConfig = !script.fileExists('ui5-deploy.yaml')
+        } else {
+            if (script.fileExists(deployConfigFile)) {
+                // in this case we will use the config file
+                useConfigFile = true
+            } else {
+                if (deployConfigFile == 'ui5-deploy.yaml') {
+                    // in this case this is most likely provided by the piper default config and
+                    // it was not explicitly configured. Hence we assume not having a config file
+                    useConfigFile = false
+                    noConfig = true
+                } else {
+                    script.error("Configured deploy config file '${deployConfigFile}' does not exists.")
+                }
+            }
+        }
+
+        if (noConfig) {
+            params += ['--noConfig'] // no config file, but we will provide our parameters
+        }
+
+        if (useConfigFile) {
+            params += ['-c', "\"" + deployConfigFile + "\""]
+        }
+
+        //
+        // All the parameters below encapsulated in an if statement might also be provided in a config file.
+        // In case they are empty we don't add to the command line and we trust in the config file.
+        // In case they are finally missing the fiori deploy toolset will tell us.
+        //
+
+        if (transportRequestId) {
+            params += ['-t', transportRequestId]
+        }
+
+        if (endpoint) {
+            params += ['-u', endpoint]
+        }
+
+        if (abapPackage) {
+            params += ['-p', abapPackage]
+        }
+
+        if (applicationName) {
+            params += ['-n' , applicationName]
+        }
+
+        if (client) {
+            params += ['-l', client]
+        }
+
+        params += ['-e', "\"" + desc + "\""]
+
+        params += ['-f'] // failfast --> provide return code != 0 in case of any failure
+
+        params += ['-y'] // autoconfirm --> no need to press 'y' key in order to confirm the params and trigger the deployment
+
+        // Here we provide the names of the environment variable holding username and password. Below we set these values.
+        params += ['--username', 'ABAP_USER', '--password', 'ABAP_PASSWORD']
+
+        def fioriDeployCmd = "fiori deploy ${params.join(' ')}"
+        script.echo "Executing deploy command: '${fioriDeployCmd}'"
+        cmd << fioriDeployCmd
+
+        script.withCredentials([script.usernamePassword(
+            credentialsId: credentialsId,
+            passwordVariable: 'password',
+            usernameVariable: 'username')]) {
+
+            /*
+                After installing the deploy toolset we switch the user. Since we do not su with option '-l' the
+                environment variables are preserved. --> The environment variables for user and password are
+                still available after the user switch.
```

Comment:
```suggestion
                After installing the deploy toolset we switch the user. Since we do not `su` with option `-l` the
                environment variables are preserved. Hence the environment variables for user and password are
                still available after the user switch.
```

Reply:
This is "only" a normal comment, not a javadoc comment. Nevertheless I'm fine also with the proposed change.